### PR TITLE
Add unit test, update publish.yaml and package.json

### PR DESCRIPTION
**Pull request by alapha23**
## Description

* Add unit tests and integration tests for fc-info for method "info" and "help"
* Update publish.yaml with new properties (regions, service name, etc.)
* Update package.json with jest dependencies
* Remove interactive credential checking due to non-interactive test environment

## Steps

```
npm install
npm test
npm run test-integration
```

## Expected output

image

### CODE REVIEW

File: test/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    console.log('sync help result: ', result);
+    expect(result).toBeUndefined();
+  });
+
+  it('info yaml', async () => {
+    const inp = _.cloneDeep(inputs);
+    inp.args = `--target-dir ${dir}`;
+    const result = await componentStarter.info(inp);
+    console.log('sync yaml result: ', result);
+    expect(JSON.stringify(result)).toMatch(/\/test\/testSync\/AccountID_cn-shenzhen_xxxxx_xxxxx/);
```
Comment (wss-git) [inline test/index.test.ts:74]: expect(Object.keys(result)).toEqual(['service', 'function', 'triggers']);

 断言要根据返回结果来预测

File: test/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    } });
+    sandbox.stub(FC.prototype, 'getFunction').resolves({
+      data: { functionName: name },
+    });
+    sandbox.stub(FC.prototype, 'getFunctionCode').resolves({
+      data: { url: 'https://registry.devsapp.cn/simple/devsapp/fc-info/zipball/0.0.11' }
+    });
+    sandbox.stub(FC.prototype, 'listTriggers').resolves({
+      data: { triggers: [{ name }] },
+    });
```
Comment (wss-git) [inline test/index.test.ts:53]: sandbox.stub(FC.prototype, 'getTrigger').resolves({
      data: {
        triggerName: 'http',
        triggerType: 'http',
        triggerConfig: {},
      },
    });

 这里缺少一个模拟接口，程序会和线上交互，然后报错 403 异常

**Comment by wss-git** [comment]
过程中不能存在交互逻辑，会阻塞ci 的进行

File: test/index.test.ts (hunk 1 above)
Comment (alapha23) [reply test/index.test.ts]: Fixed in latest commit. Thank you.

File: test/index.test.ts (hunk 2 above)
Comment (alapha23) [reply test/index.test.ts:53]: Fixed. Thank you very much.

File: publish.yaml [hunk 3]
```diff
@@ -1,10 +1,58 @@
...
+    Type:
+      - String
+  functionName:
+    Description: 函数名称
+    Required: false
+    Example: function-name
+    Default: ''
+    Type:
+      - String
+  targetDir:
```
Comment (wss-git) [inline publish.yaml:52]: sync 不接受这个参数吧

File: package.json [hunk 4]
```diff
@@ -26,6 +26,8 @@
     "start": "npm run watch",
     "watch": "npm run doc && tsc -w",
     "build": "rm -rf node_modules && rm -rf package-lock.json && npm i && ncc build src/index.ts -m -e @serverless-devs/core -o lib",
+    "test": "jest",
```
Comment (wss-git) [inline package.json:29]: jest --testNamePattern '^(?!Integration::)'

File: package.json [hunk 5]
```diff
@@ -26,6 +26,8 @@
...
+    "test:cov": "jest --coverage",
```
Comment (wss-git) [inline package.json:30]: jest --testNamePattern ^Integration::

Comment (alapha23) [reply package.json:30]: 请问`jest --testNamePattern ^Integration::`指令会不会触发集成测试阿，这里`test:cov`应该是触发coverage？

File: publish.yaml (hunk 3 above)
Comment (alapha23) [reply publish.yaml]: Fixed.

File: package.json (hunk 4 above)
Comment (alapha23) [reply package.json]: Fixed

File: test/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+describe('test/index.test.ts', () => {
+  beforeEach(async () => {
+    sandbox.stub(FC.prototype, 'getService').resolves({
+      data: { name },
+    });
+    sandbox.stub(FC.prototype, 'listFunctions').resolves({
+      data: { functions: [{ name }],
+    } });
+    sandbox.stub(FC.prototype, 'getFunction').resolves({
+      data: { functionName: name },
```
Comment (git-qfzhang) [inline test/index.test.ts:46]: getFunction 是否应该把 function 的一些必填配置加上，例如 memorySize，handler 这些，即便是 mock 的只返回一个 name 也太简洁了吧

File: test/index.test.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      data: { url: 'https://registry.devsapp.cn/simple/devsapp/fc-info/zipball/0.0.11' }
+    });
+    sandbox.stub(FC.prototype, 'listTriggers').resolves({
+      data: { triggers: [{ name }] },
+    });
+    sandbox.stub(FC.prototype, 'getTrigger').resolves({
+      data: {
+        triggerName: 'http',
+        triggerType: 'http',
+        triggerConfig: {},
```
Comment (git-qfzhang) [inline test/index.test.ts:58]: triggerConfig 也可以填充一下，可以只填充最简单的配置，但是不要留空

File: test/index.test.ts [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+      },
+    });
+  });
+
+  afterEach(async () => {
+    sandbox.restore();
+    await fse.remove(dir);
+  });
+
+  it('info info', async () => {
```
Comment (git-qfzhang) [inline test/index.test.ts:68]: 同上

File: test/index.test.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    const inp = _.cloneDeep(inputs);
+    inp.args = 'info';
+    const result = await componentStarter.info(inp);
+    expect(Object.keys(result)).toEqual(['service', 'function', 'triggers']);
```
Comment (git-qfzhang) [inline test/index.test.ts:72]: 同上

File: test/index-integration.test.ts [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+          code: {
+            zipFile,
+	  },
+    });
+ 
+    const inp = _.cloneDeep(inputs);
+    inp.args = 'info';
+    const componentStarter = new ComponentStarter();
+    const result = await componentStarter.info(inp);
+    expect(Object.keys(result)).toEqual(['service','function',]);
```
[the remark this answers is not in the document]
Comment (alapha23) [reply test/index-integration.test.ts:87]: @wss-git 建议只判断key

File: test/index-integration.test.ts [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  const funcName = `func-${new Date().getTime()}-${Math.random().toString(36).substr(2)}`
+  //const triggerName = `trigger-${new Date().getTime()}-${Math.random().toString(36).substr(2)}`
+
+  const inputs = {
+    props: {
+      region: 'cn-shenzhen',
+      serviceName: serviceName,
+      functionName: funcName,
+    },
+    credentials: {
```
[the remark this answers is not in the document]
Comment (alapha23) [reply test/index-integration.test.ts:23]: 这里不传credentials的话，`npm run test` 会要求选择 access account
image

File: test/index-integration.test.ts [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+      AccessKeySecret: process.env.AccessKeySecret,
+    },
+    appName: 'fc-info-test',
+    project: {
+      component: 'devsapp/fc-info',
+      access: 'test',
+      projectName: 'test',
+    },
+    command: '',
+    args: '--region',
```
[the remark this answers is not in the document]
Comment (alapha23) [reply test/index-integration.test.ts:35]: 这个是参考 fc-sync加的，这样我把它删了

File: package.json (hunk 5 above)
Comment (alapha23) [reply package.json:34]: @wss-git

File: test/index.test.ts (hunk 6 above)
Comment (alapha23) [reply test/index.test.ts]: Fixed

File: test/index.test.ts (hunk 7 above)
Comment (alapha23) [reply test/index.test.ts]: Fixed. How does it look now?

File: .github/workflows/ci.yml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+      - name: unit test 
+        run: |
+          npm run test
+      - name: integration test
+        env:
+            AccountID: ${{secrets.AccountID}}
+            AccessKeyID: ${{secrets.AccessKeyID}}
+            AccessKeySecret: ${{secrets.AccessKeySecret}}
+        run: |
+          npm run test-integration
```
Comment (wss-git) [inline .github/workflows/ci.yml:38]: env 修改为：
secrets.AccountID || secrets.ACCOUNTID
secrets.AccessKeyID || secrets.ACCESSKEYID
secrets.AccessKeySecret || secrets.ACCESSKEYSECRET
下面这个 run 这样写试试呢
sudo npm run test-integration

**Comment by git-qfzhang** [comment]
可以增加 corner case
